### DISCUSSION TITLE
Local Session Save vs Download

### DISCUSSION THREAD

**Discussion by lreading** [OWASP/threat-dragon#384] [Q&A] [answered]
Having a local session option for Threat Dragon is new in 2.0.  Previously, "save" has always been a seamless operation without any kind of prompt for the user.  With local session, we will need to download a copy of the threat model json file. 

There are currently 2 places where you can save a threat model:
- The high level edit page
- The graph edit page

Do we want "save" to persist changes in memory, the same as it has always worked?
If so, do we add a separate download button?  Where?
Do we want "save" to prompt a download each time it's clicked?

**Comment by jgadsden** [comment, marked as answer]
I am working on this with issue #383 and the thinking at the moment is to avoid the "save" to prompt a download each time Save is clicked. I think we can get away without a download button, but we do need an 'open' button when in local mode. I added this with pull-request #535 , but this functionality is incomplete (does not work with Firefox for example)

**Reply by jgadsden** [reply]
implemented by Open/Save/Rename for Electron #383
